tests/Unit/Evolution/Systems/GeneralizedHarmonic/TestFunctions.py

```diff
@@ -184,3 +183,199 @@ def two_index_constraint(d_gauge_function, spacetime_normal_one_form,
     return constraint
 
 # End test functions for two-index constraint
+
+
+# Begin tests for Test_DampedHarmonicGaugeQuantities.cpp
+
+def weight_func(coords, r_max, r_width):
+    r2 = np.sum([coords[i]**2 for i in range(len(coords))])
+    return np.exp(- r2 * r_width / r_max / r_max)
+
+
+def spacetime_deriv_weight_func(coords, r_max, r_width):
+    W = weight_func(coords, r_max, r_width)
+    DW = np.zeros(len(coords) + 1)
+    DW[1:] = -2. * r_width * W / r_max / r_max * coords
+    return DW
+
+
+def roll_on_func(time, t_start, sigma_t):
+    if time < t_start: return 0.
+    return 1. - np.exp(- ((time - t_start)/sigma_t)**4 )
+
+
+def time_deriv_roll_on_func(time, t_start, sigma_t):
+    if time < t_start: return 0.
+    tnrm = (time - t_start)/sigma_t
+    return np.exp(- tnrm**4) * 4 * tnrm**3 / sigma_t
+
+def log_fac(lapse, sqrt_det_spatial_metric, exponent):
+    # if exponent == 0, the first term automatically vanishes
+    return exponent * np.log(sqrt_det_spatial_metric**2) - np.log(lapse)
+
+def spacetime_deriv_log_fac(lapse, shift, spacetime_unit_normal,
+        inverse_spatial_metric, sqrt_det_spatial_metric, dt_spatial_metric,
+        pi, phi, exponent):
+    SpatialDim = len(shift)
+    #
+    import os, imp
+    dname = os.path.abspath(os.path.join(os.path.dirname(\
+                os.path.realpath(__file__)),\
+                '../../../PointwiseFunctions/GeneralRelativity/'))
+    if not os.path.exists(dname):
+        dname = os.path.abspath(os.path.join(os.path.dirname(\
+                os.path.realpath(__file__)),\
+                '../../PointwiseFunctions/GeneralRelativity/'))
+    fname = os.path.join(dname, 'TestFunctions.py')
+    TF = imp.load_source('TestFunctions', fname)
+    #
+    detg = sqrt_det_spatial_metric**2
+    dg = TF.spacetime_deriv_detg(sqrt_det_spatial_metric,
+            inverse_spatial_metric, dt_spatial_metric, phi)
+    d0N = TF.dt_lapse(lapse, shift, spacetime_unit_normal, phi, pi)
+    d3N = TF.deriv_lapse(lapse, spacetime_unit_normal, phi)
+    d4N = np.zeros(1 + SpatialDim)
+    d4N[0]  = d0N
+    d4N[1:] = d3N
+    # if exponent == 0, the first term automatically vanishes
+    d_logfac = (exponent / detg) * dg - (1. / lapse) * d4N
+    return d_logfac
+
+def spacetime_deriv_pow_log_fac(lapse, shift, spacetime_unit_normal,
+        inverse_spatial_metric, sqrt_det_spatial_metric, dt_spatial_metric,
+        pi, phi, g_exponent, exponent):
+    exponent = int(exponent)
+    dlogfac = spacetime_deriv_log_fac(lapse, shift, spacetime_unit_normal,
+        inverse_spatial_metric, sqrt_det_spatial_metric, dt_spatial_metric,
+        pi, phi, g_exponent)
+    if exponent == 1:
```

Comment:
Here you don't need the if/else control flow, because performance is not relevant in tests.
Or do you need this here to maintain a certain level of precision?